docs/expressions.md

```diff
@@ -0,0 +1,89 @@
+'Andrew said: "Hello!"'
+```
+
+### Identifiers
+Identfier is a variable, which adressed by name. QL supports the following identifiers:
+- `logID` - the log unique identifier.
+- `ctime` - the record created time (every record gets its ctime when it is added to the log). For `ctime` only the `<` and `>` operations are allowed.
+
+### Functions
+A function is a value that is calculated from the arguments provided. It looks like an identifier followed by arguments in parentheses. The argument list may be empty.
+
+Solaris supports the following functions:
+- `tag(<name>)` - returns the tag value for a log. Name could be a string constant or any other argument value
+
+### List of constants
+Some constants maybe groupped in a list. The List defined like the coma-separted constants in between `[` and `]`:
+
+```
+["a", "b", 'c'] 
+```
+the list of three string constants - "a", "b" and "c"
+
+## Operations
+Operation is an action which requires two arguments. All operations return either TRUE or FALSE
+
+Examples:
+```
+'1234' = '234' // compares two string constants, the result will be FALSE
+logID != "123" // compares log ID with the string "123", the result depends on the logID value
+tag("t1") > tag("t2") // compares value of the tag t1 with the value of the tag t2, the result depends on the tags values
+tag("t1") IN ["1", "2", "3"] // the value of t1 is either "1", "2", or "3"  
+```
+
+QL supports the following operations:
+
+"<", ">", "<=", ">=", "!=", "="
+
+| Operation | Description                                                                                                 |
+|-----------|-------------------------------------------------------------------------------------------------------------|
+| <         | The left argument is less than the right one                                                                |
+| >         | The left argument is greater than the right one                                                             |
+| <=        | The left argument is less or equal to the right one                                                         |
+| >=        | The left argument is greater or equal to the right one                                                      |
+| !=        | The left argument is not equal to the right one                                                             |
+| =         | The left argument is equal to the right one                                                                 |
+| IN        | The left argument value is in the list. Right argument must be a list                                       |
+| LIKE      | The left argument should be like the constant (second argument). The operation is similart to the SQL like. |
+
+## QL boolen expression
+The QL expression is the series of boolean values that can be combined by AND, OR, NOT boolean operations and the parenthesis to increase the priority.
+
+Examples:
+```
+tag('t1') != tag('t2') 
+ctime > "2024-02-12 00:00:00.000"
```

Comment:
ctime > "2024-02-12 00:00:00.000" AND ctime < "2024-03-12 00:00:00.000"

Reply:
done